src/core/algorithms/ind/faida/inclusion_testing/sampled_inverted_index.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <atomic_bitvector.hpp>
+#include <hash_set2.hpp>
+#include <hash_table7.hpp>
+#include <hash_table8.hpp>
+#include <mutex>
+#include <unordered_map>
+
+#include <boost/dynamic_bitset.hpp>
+
+#include "algorithms/ind/faida/util/simple_ind.h"
+#include "hll_data.h"
+
+namespace algos::faida {
+
+class SampledInvertedIndex {
+private:
+    struct SpinLock {
```

Comment:
Have you tried using a spinlock from `boost`? @polyntsov is sure that writing a custom spinlock is not worth it.
If this implementation of a spinlock is faster than a `boost`, then you need to attach the results of experiments and explicitly move this implementation into the `util` directory

Reply:
No, I have not tried it. In this PR I will not perform any experiments, so if you think this piece of code somehow impacts the code quality, I will just remove it. However, I still do not see any problems with leaving it here, since this is just an experimental feature located in the local code and related only to Faida. If in the future any programmer will want to use  spinlock, it may use one from boost himself.